cmd/river/go.mod

```diff
@@ -1,23 +1,23 @@
 module github.com/riverqueue/river/cmd/river
 
-go 1.22.5
+go 1.21.13
 
-// replace github.com/riverqueue/river => ../..
+toolchain go1.22.5
 
-// replace github.com/riverqueue/river/riverdriver => ../../riverdriver
+replace github.com/riverqueue/river => ../..
 
-// replace github.com/riverqueue/river/riverdriver/riverdatabasesql => ../../riverdriver/riverdatabasesql
+replace github.com/riverqueue/river/riverdriver => ../../riverdriver
 
-// replace github.com/riverqueue/river/riverdriver/riverpgxv5 => ../../riverdriver/riverpgxv5
+replace github.com/riverqueue/river/riverdriver/riverdatabasesql => ../../riverdriver/riverdatabasesql
 
-// replace github.com/riverqueue/river/rivertype => ../../rivertype
+replace github.com/riverqueue/river/riverdriver/riverpgxv5 => ../../riverdriver/riverpgxv5
+
+replace github.com/riverqueue/river/rivertype => ../../rivertype
```

Comment:
This'll need the two stage release. Without these `replace` directives, `go mod tidy` detects that Go 1.21 can't be supported and therefore removes the `toolchain` directive and changes `go` from 1.21 to 1.22.

`replace` directives to be removed during second phase of release.

Reply:
Is this why setup-go is still upgrading us to 1.22?

```
Setup go version spec 1.21
Found in cache @ /opt/hostedtoolcache/go/1.21.[12](https://github.com/riverqueue/river/actions/runs/10312078359/job/28546717441#step:4:13)/x64
Added go to the path
Successfully set up Go version 1.21
go: downloading go1.22.5 (linux/amd64)
/opt/hostedtoolcache/go/1.21.12/x64/bin/go env GOMODCACHE
/opt/hostedtoolcache/go/1.21.12/x64/bin/go env GOCACHE
/home/runner/go/pkg/mod
/home/runner/.cache/go-build
Cache Size: ~1[16](https://github.com/riverqueue/river/actions/runs/10312078359/job/28546717441#step:4:17) MB (122068102 B)
/usr/bin/tar -xf /home/runner/work/_temp/3fd374b5-c372-4fe0-bb5a-72c9ec40afc9/cache.tzst -P -C /home/runner/work/river/river --use-compress-program unzstd
Received 12[20](https://github.com/riverqueue/river/actions/runs/10312078359/job/28546717441#step:4:21)68102 of 122068102 (100.0%), 116.4 MBs/sec
Cache restored successfully
Cache restored from key: setup-go-Linux-ubuntu22-go-1.22.5-7e42e876c7fc43348578ac5ddfec93c5643eb4a8c880ab849c10c6fbcbbb6cc8
go version go1.[22](https://github.com/riverqueue/river/actions/runs/10312078359/job/28546717441#step:4:23).5 linux/amd64
```

Reply:
I think that might be something else -- even Go 1.21 is capable of upgrading to a Go 1.22 toolchain, so the default `GOTOOLCHAIN` value of `auto` is causing Go to install the matrix version, but then fetch one according to `toolchain` in `go.mod`.